src/main/java/org/opentestsystem/ap/ims/service/AttachmentService.java

```diff
@@ -16,43 +16,196 @@
+            archiveUpload.getFile().transferTo(giArchive.toFile());
+
+            // Upload archive zip
+            // TODO: remove this if/when GI logic no longer relies on zip archive
+            contentModel.setContentZipFile(uploadCoreFile(
+                    archiveUpload.getItemId(),
+                    userTransaction.getSection(),
+                    archiveUpload.getFile().getOriginalFilename(),
+                    giArchive));
+
+            // Unzip, validate, and upload gi archive contents
+            giContents = Files.createTempDirectory("gi-contents");
+            ZipUtil.unzip(giArchive, giContents);
+
+            final Map<String, Path> uploadedResources = FileUtils.listFiles(giContents.toFile(),null,false).stream()
+                    .collect(Collectors.toMap(
+                            File::getName,
+                            File::toPath
+                    ));
+
+            // Locate the uploaded GAX file
+            final List<Path> gaxFiles = uploadedResources.entrySet().stream()
+                    .filter(entry -> FilenameUtils.isExtension(entry.getKey(), "gax"))
+                    .map(Map.Entry::getValue)
+                    .collect(Collectors.toList());
+            if (gaxFiles.size() != 1) {
+                throw new ValidationException("GI Archive must contain one and only one .gax GI definition XML file");
+            }
+            final Path gaxFile = gaxFiles.get(0);
+
+            // Save GAX / GI definition in item.json
+            try (final InputStream gaxStream = Files.newInputStream(gaxFile)) {
+                contentModel.setGiDefinition(IOUtils.toString(gaxStream, StandardCharsets.UTF_8).trim());
+            }
+
+            // Validate and upload GAX-referenced resources
+            final List<String> referencedFilenames = AttachmentSelector.getReferencedGaxFileNames(gaxFile.toFile());
+            final List<Attachment> referencedResources = referencedFilenames.stream()
+                    .map(referencedFilename -> {
+                        if (!uploadedResources.containsKey(referencedFilename)) {
+                            throw new ValidationException(format("GAX references missing resource: %s", referencedFilename));
+                        }
+                        return uploadedResources.get(referencedFilename);
+                    })
+                    .map(resource -> uploadCoreFile(
+                            archiveUpload.getItemId(),
+                            userTransaction.getSection(),
+                            resource.getFileName().toString(),
+                            resource
+                    ))
+                    .collect(Collectors.toList());
+
+            // Include translated GAX resource overrides
+            // TODO These should probably be represented by the UI and model as translation resource overrides
+            //  rather than reply on zip-contents file-naming conventions.
```

Comment:
Any GI resource can have a localized override version by including an additional resource in the zip appended with "_ESN"
For example:
Original: `my-background-image.png`
Spanish override: `my-background-image_ESN.png`